File: php/features.php
<?php
declare(strict_types=1);

// CatFact SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CatFactFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CatFactBaseFeature();
            case "test":
                return new CatFactTestFeature();
            default:
                return new CatFactBaseFeature();
        }
    }
}

File: py/features.py
# CatFact SDK feature factory

from feature.base_feature import CatFactBaseFeature
from feature.test_feature import CatFactTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CatFactBaseFeature(),
        "test": lambda: CatFactTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
